File: apps/php/src/FizzBuzz.php
<?php

namespace App;

use App\Domain\FizzBuzzException;
use App\Domain\Type\FizzBuzzType;
use App\Domain\Type\FizzBuzzTypeName;

class FizzBuzz
{
    public static function create(int $type): FizzBuzzType
    {
        return FizzBuzzTypeName::tryFrom($type)?->createType()
            ?? throw new FizzBuzzException("タイプ{$type}は見つかりません");
    }

    public function generate(int $number): string
    {
        $type = self::create(1);

        return $type->generate($number)->getValue();
    }

    /**
     * @return string[]
     */
    public function generateList(): array
    {
        $result = [];
        for ($i = 1; $i <= 100; $i++) {
            $result[] = $this->generate($i);
        }

        return $result;
    }

    public function printFizzBuzz(): void
    {
        $list = $this->generateList();
        foreach ($list as $item) {
            echo $item . "\n";
        }
    }
}

File: apps/rust/src/fizz_buzz.rs
use std::io::Write;

use crate::domain::types::{FizzBuzzType, FizzBuzzType01, FizzBuzzTypeName};

pub fn create(type_number: i32) -> Result<Box<dyn FizzBuzzType>, String> {
    Ok(FizzBuzzTypeName::from_number(type_number)?.create_type())
}

pub fn generate(number: i32) -> String {
    FizzBuzzType01.generate(number).to_string()
}

pub fn generate_list(start: i32, end: i32) -> Vec<String> {
    (start..=end).map(generate).collect()
}

pub fn print_fizzbuzz(writer: &mut dyn Write) {
    for s in generate_list(1, 100) {
        writeln!(writer, "{}", s).unwrap();
    }
}
